### ISSUE TITLE
canny_bp either can't back propagation 

### ISSUE THREAD

**Issue by qzq2514** [GuangyuanHao/canny-bp-tensorflow#1]
Thanks for your codes. Now I want to train a GAN. 
I use canny_bp.py to detect edge of images and calculate L1 loss between orginal image and generated image, it still can't  back propagate to update the network(generatot).
Have you really used canny_bp.py to calculate loss in order to update parameters.
PS:Of course, canny_bp.py and canny.py all can detect edge of images. Nice code~

**Comment by GuangyuanHao** [comment]
Yes, I have used. It should work. The logit of code is clear and you may change it to adapt your algorithm based on my code.
